pkg/reconciler/kubernetes/tektoninstallerset/install.go

```diff
@@ -183,167 +183,239 @@ func (i *installer) EnsureNamespaceScopedResources() error {
 
 func (i *installer) EnsureDeploymentResources() error {
 	for _, d := range i.deployment {
-		if err := i.ensureDeployment(&d); err != nil {
+		if err := i.ensureResource(&d); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func computeDeploymentHash(d appsv1.Deployment) (string, error) {
-	// set replicas to a constant value and then calculate hash so
-	// that later if user updates replicas, we can exclude that change.
-	// setting the replicas to same const and checking the hash
-	// so that we can allow only replica change revert any other change
-	// done to the deployment spec
-	d.Spec.Replicas = ptr.Int32(replicasForHash)
-
-	return hash.Compute(d.Spec)
-}
-
-func (i *installer) createDeployment(expected *unstructured.Unstructured) error {
-
-	dep := &appsv1.Deployment{}
-	err := runtime.DefaultUnstructuredConverter.FromUnstructured(expected.Object, dep)
-	if err != nil {
-		return err
-	}
-
-	hash, err := computeDeploymentHash(*dep)
-	if err != nil {
-		return fmt.Errorf("failed to compute hash of deployment: %v", err)
-	}
-
-	if len(dep.Annotations) == 0 {
-		dep.Annotations = map[string]string{}
-	}
-	dep.Annotations[v1alpha1.LastAppliedHashKey] = hash
-
-	unstrObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(dep)
-	if err != nil {
-		return err
-	}
-	expected.SetUnstructuredContent(unstrObj)
-
-	return i.mfClient.Create(expected)
-}
-
-func (i *installer) updateDeployment(existing *unstructured.Unstructured, existingDeployment, expectedDeployment *appsv1.Deployment) error {
-	i.logger.Infof("updating resource %s: %s/%s", existing.GetKind(), existing.GetNamespace(), existing.GetName())
-
-	// save on cluster replicas in a var and assign it back to deployment
-	onClusterReplicas := existingDeployment.Spec.Replicas
-
-	existingDeployment.Spec = expectedDeployment.Spec
-	existingDeployment.Spec.Replicas = onClusterReplicas
-
-	// compute new hash of spec and add as annotation
-	newHash, err := computeDeploymentHash(*existingDeployment)
-	if err != nil {
-		return fmt.Errorf("failed to compute new hash of existing deployment: %v", err)
-	}
-
-	if len(existingDeployment.Annotations) == 0 {
-		existingDeployment.Annotations = map[string]string{}
-	}
-
-	existingDeployment.Annotations[v1alpha1.LastAppliedHashKey] = newHash
-
-	unstrObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(existingDeployment)
-	if err != nil {
-		return err
-	}
-	existing.SetUnstructuredContent(unstrObj)
+// list of fields should be reconciled
+func (i *installer) resourceReconcileFields(u *unstructured.Unstructured) []string {
+	switch u.GetKind() {
+	case "Deployment", "StatefulSet":
+		return []string{
+			annotationsPath,
+			labelsPath,
+			"spec",
+		}
```

Comment:
with this PR change still user can modify metadata of deployments manually and i think even that should not be allowed to user to modify manually

I tried below way and it works for me
```suggestion
		return []string{
			annotationsPath,
			labelsPath,
			"metadata",
			"spec",
		}
```

or something differently need to be handled :thinking:

Reply:
@savitaashture under `metadata` monitoring only the `annotations` and `labels` that too not all the labels and annotations.
k8s keeps some of the annotations of it's own reference. So we can not match 1:1
however I have changes the code to keep the labels and annotations from tektonConfigCR as is.

Reply:
annotationsPath => `metadata.annotations`
labelsPath => `metadata.labels`

Reply:
so that means labels and annotations of metadata can be changed directly by user am i right ?

Reply:
@savitaashture user/services can add extra labels or annotations. However they can't change the defined[1] label/annotation values

[1] - values from manifests and tecktonConfig CR